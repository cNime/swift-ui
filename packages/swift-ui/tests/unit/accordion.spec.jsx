import { mount } from '@vue/test-utils'
import Accordion from '../../src/components/accordion/index.vue'

const factory = (values = {}, slots = { default: ['按钮'] }) => {
  return mount(Accordion, {
    slots,
    props: { ...values }
  })
}

describe('Accordion Snap', () => {
  it('render initial Accordion', () => {
    const wrapper = mount(Accordion)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props title', () => {
    const wrapper = mount(<Accordion title="title" />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props open', () => {
    const wrapper = mount(<Accordion open={true} />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props icon', () => {
    const wrapper = factory(
      {
        icon: { value: 'chevron-down', color: 'red' }
      },
      {
        default: ['<div></div>']
      }
    )
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props icon prefixClass', () => {
    const wrapper = factory(
      {
        icon: { prefixClass: 'prefixClass', value: 'star', color: 'red' }
      },
      {
        default: ['<div></div>']
      }
    )
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props note', () => {
    const wrapper = factory(
      {
        note: 'note'
      },
      {
        default: ['<div></div>']
      }
    )
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props note', () => {
    const wrapper = factory(
      {
        note: 'note'
      },
      {
        default: ['<div></div>']
      }
    )
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Accordion Event', () => {
  it('toggle accordion', () => {
    const wrapper = factory()
    wrapper.find('.at-accordion__header').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('click')
  })
})
