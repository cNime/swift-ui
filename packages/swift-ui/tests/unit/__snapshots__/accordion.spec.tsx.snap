// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Snap render Accordion -- props icon 1`] = `
<div
  class="at-accordion"
>
  <div
    class="at-accordion__header"
  >
    <div
      class="at-icon at-icon-chevron-down at-accordion__icon"
      style="color: red; font-size: undefinedpx;"
    />
    <div
      class="at-accordion__info"
    >
      <div
        class="at-accordion__info__title"
      >
        
      </div>
      <div
        class="at-accordion__info__note"
      >
        
      </div>
    </div>
    <div
      class="at-accordion__arrow"
    >
      <div
        class="at-icon at-icon-chevron-down"
      />
    </div>
  </div>
  <div
    class="at-accordion__content at-accordion__content--inactive"
  >
    <div
      class="at-accordion__body"
    >
      
      &lt;div&gt;&lt;/div&gt;
      
    </div>
  </div>
</div>
`;

exports[`Accordion Snap render Accordion -- props icon prefixClass 1`] = `
<div
  class="at-accordion"
>
  <div
    class="at-accordion__header"
  >
    <div
      class="prefixClass prefixClass-star at-accordion__icon"
      style="color: red; font-size: undefinedpx;"
    />
    <div
      class="at-accordion__info"
    >
      <div
        class="at-accordion__info__title"
      >
        
      </div>
      <div
        class="at-accordion__info__note"
      >
        
      </div>
    </div>
    <div
      class="at-accordion__arrow"
    >
      <div
        class="at-icon at-icon-chevron-down"
      />
    </div>
  </div>
  <div
    class="at-accordion__content at-accordion__content--inactive"
  >
    <div
      class="at-accordion__body"
    >
      
      &lt;div&gt;&lt;/div&gt;
      
    </div>
  </div>
</div>
`;

exports[`Accordion Snap render Accordion -- props isAnimation 1`] = `
<div
  class="at-accordion"
>
  <div
    class="at-accordion__header"
  >
    
    <div
      class="at-accordion__info"
    >
      <div
        class="at-accordion__info__title"
      >
        
      </div>
      <div
        class="at-accordion__info__note"
      >
        
      </div>
    </div>
    <div
      class="at-accordion__arrow at-accordion__arrow--folded"
    >
      <div
        class="at-icon at-icon-chevron-down"
      />
    </div>
  </div>
  <div
    class="at-accordion__content"
  >
    <div
      class="at-accordion__body"
    >
      <!---->
    </div>
  </div>
</div>
`;

exports[`Accordion Snap render Accordion -- props note 1`] = `
<div
  class="at-accordion"
>
  <div
    class="at-accordion__header"
  >
    
    <div
      class="at-accordion__info"
    >
      <div
        class="at-accordion__info__title"
      >
        
      </div>
      <div
        class="at-accordion__info__note"
      >
        note
      </div>
    </div>
    <div
      class="at-accordion__arrow"
    >
      <div
        class="at-icon at-icon-chevron-down"
      />
    </div>
  </div>
  <div
    class="at-accordion__content at-accordion__content--inactive"
  >
    <div
      class="at-accordion__body"
    >
      
      &lt;div&gt;&lt;/div&gt;
      
    </div>
  </div>
</div>
`;

exports[`Accordion Snap render Accordion -- props note 2`] = `
<div
  class="at-accordion"
>
  <div
    class="at-accordion__header"
  >
    
    <div
      class="at-accordion__info"
    >
      <div
        class="at-accordion__info__title"
      >
        
      </div>
      <div
        class="at-accordion__info__note"
      >
        note
      </div>
    </div>
    <div
      class="at-accordion__arrow"
    >
      <div
        class="at-icon at-icon-chevron-down"
      />
    </div>
  </div>
  <div
    class="at-accordion__content at-accordion__content--inactive"
  >
    <div
      class="at-accordion__body"
    >
      
      &lt;div&gt;&lt;/div&gt;
      
    </div>
  </div>
</div>
`;

exports[`Accordion Snap render Accordion -- props open 1`] = `
<div
  class="at-accordion"
>
  <div
    class="at-accordion__header"
  >
    
    <div
      class="at-accordion__info"
    >
      <div
        class="at-accordion__info__title"
      >
        
      </div>
      <div
        class="at-accordion__info__note"
      >
        
      </div>
    </div>
    <div
      class="at-accordion__arrow at-accordion__arrow--folded"
    >
      <div
        class="at-icon at-icon-chevron-down"
      />
    </div>
  </div>
  <div
    class="at-accordion__content"
  >
    <div
      class="at-accordion__body"
    >
      <!---->
    </div>
  </div>
</div>
`;

exports[`Accordion Snap render Accordion -- props title 1`] = `
<div
  class="at-accordion"
>
  <div
    class="at-accordion__header"
  >
    
    <div
      class="at-accordion__info"
    >
      <div
        class="at-accordion__info__title"
      >
        title
      </div>
      <div
        class="at-accordion__info__note"
      >
        
      </div>
    </div>
    <div
      class="at-accordion__arrow"
    >
      <div
        class="at-icon at-icon-chevron-down"
      />
    </div>
  </div>
  <div
    class="at-accordion__content at-accordion__content--inactive"
  >
    <div
      class="at-accordion__body"
    >
      <!---->
    </div>
  </div>
</div>
`;

exports[`Accordion Snap render initial Accordion 1`] = `
<div
  class="at-accordion"
>
  <div
    class="at-accordion__header"
  >
    
    <div
      class="at-accordion__info"
    >
      <div
        class="at-accordion__info__title"
      >
        
      </div>
      <div
        class="at-accordion__info__note"
      >
        
      </div>
    </div>
    <div
      class="at-accordion__arrow"
    >
      <div
        class="at-icon at-icon-chevron-down"
      />
    </div>
  </div>
  <div
    class="at-accordion__content at-accordion__content--inactive"
  >
    <div
      class="at-accordion__body"
    >
      <!---->
    </div>
  </div>
</div>
`;

exports[`Accordion Snap test change props open 1`] = `
<div
  class="at-accordion"
>
  <div
    class="at-accordion__header"
  >
    
    <div
      class="at-accordion__info"
    >
      <div
        class="at-accordion__info__title"
      >
        
      </div>
      <div
        class="at-accordion__info__note"
      >
        
      </div>
    </div>
    <div
      class="at-accordion__arrow at-accordion__arrow--folded"
    >
      <div
        class="at-icon at-icon-chevron-down"
      />
    </div>
  </div>
  <div
    class="at-accordion__content"
    style="height: 0px;"
  >
    <div
      class="at-accordion__body"
    >
      
      按钮
      
    </div>
  </div>
</div>
`;
